micrometer-core/src/main/java/io/micrometer/core/instrument/distribution/AbstractTimeWindowHistogram.java

```diff
@@ -45,7 +44,7 @@ abstract class AbstractTimeWindowHistogram<T, U> implements Histogram {
 
     private final Clock clock;
 
-    private final boolean supportsAggregablePercentiles;
+    protected final boolean supportsAggregablePercentiles;
```

Comment:
I needed to make this `protected` for `TimeWindowPercentileHistogram` to include the logic that was previously here in `takeCountSnapshot()`

Reply:
In the latest iteration, I was able to remove this field because it is no longer used anywhere but in constructors. This will save a whole byte per histogram potentially (ignoring object layout losses).